src/lib/components/select/QSelect.svelte

```diff
@@ -168,9 +175,8 @@
         <a
           href={multiple ? "javascript:void(0)" : undefined}
           class="q-select__option {selectedOptions[idx] ? 'q-select__option--selected' : ''}"
-          on:mousedown={() => (preventClose = true)}
-          on:click={(e) => select(e, option)}
-          >{typeof option === "string" ? option : option.value}</a
+          onmousedown={() => (preventClose = true)}
+          onclick={(e) => select(e, option)}>{typeof option === "string" ? option : option.value}</a
```

Comment:
You mean the user's event handlers and you're right. I gotta read up on how to do that properly in Svelte. Not sure about your `props` approach (yet).
But this is also somewhat out of scope tbh, since we're just refactoring to Svelte 5 here.